addons/web_editor/static/tests/field_html_tests.js

```diff
@@ -126,6 +132,30 @@ QUnit.module('web_editor', {}, function () {
             form.destroy();
         });
 
+        QUnit.test('check if required field is set', async function (assert) {
+            assert.expect(1);
+
+            var form = await testUtils.createView({
+                View: FormView,
+                model: 'note.note',
+                data: this.data,
+                debug: true,
+                arch: '<form>' +
+                        '<field name="header" widget="html" style="height: 100px" />' +
+                      '</form>',
+                res_id: 1,
+            });
+
+            await testUtils.form.clickEdit(form);
+            await testUtils.nextTick();
+            await testUtils.dom.click(form.$('.o_form_button_save'));
+            await testUtils.nextTick();
+
+            assert.strictEqual(form.mode, 'edit', 'should be still in edit mode if the save fails');
```

Comment:
you can actually check that it sent a warning (by intercepting event 'call_service', 'notification' service)